pkg/controller/upload_controller.go

```diff
@@ -272,13 +303,26 @@ func (c *uploadController) processUpload(req *pluginv1api.Upload) error {
 
 	_, err = c.dataMover.CopyToRepo(peID)
 	if err != nil {
-		errMsg := fmt.Sprintf("Failed to upload snapshot, %v, to durable object storage. %v", peID.String(), errors.WithStack(err))
-		_, err = c.patchUploadByStatus(req, pluginv1api.UploadPhaseUploadError, errMsg)
-		if err != nil {
-			errMsg = fmt.Sprintf("%v. %v", errMsg, errors.WithStack(err))
+		log.Infof("CopyToRepo Error Received: %v", err.Error())
+		// Check if the request was canceled.
+		// context canceled error is wrapped in awserr, hence a string compare.
+		if strings.Contains(err.Error(), context.Canceled.Error()) {
```

Comment:
It is not recommended to do error type checking using string comparison. The following reference might be useful.

https://stackoverflow.com/questions/47179024/how-to-check-if-a-request-was-cancelled

Reply:
I agree, but s3 APIs do something weird with the RequestCancelled error, they wrap it up in their own "aswerr" type. I assumed doing errors.Is(err, context.Canceled) would work, but that doesn't either. The error string compare was a last resort.